src/main/java/com/squareup/javapoet/CodeBlock.java

```diff
@@ -360,6 +363,10 @@ private TypeName argToType(Object o) {
       throw new IllegalArgumentException("expected type but was " + o);
     }
 
+    private Character argToChar(Object o) {
+      return o != null ? ((char) o) : null;
```

Comment:
(Apparently GitHub lost this one)

Check `instanceof Character` and throw `IllegalArgumentException` like above?

Reply:
Good point, should be fixed now.